### PR TITLE
Document container privilege configuration

### LINKED ISSUE #1097: Document the privilege requirements of the various images
We need to specify which images need to be privileged, and why, and which SAs need SCCs on OpenShift, and why. If any images are found to be currently more privileged than they need to be, other issues will be filed to make sure we reduce those privileges.

### PR DESCRIPTION
**Pull request by skitt**
This covers the current settings, without explaining the underlying
requirements in all cases. The gateway, route agent, and Globalnet
pods probably can have their privileges reduced; the documentation
will be updated with the results of the corresponding investigations.

Fixes: submariner-io/submariner#1097
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by mangelajo — APPROVED** [review]
Content looks good, although the fight with the linters isn't over ':D

**Review by nyechiel — APPROVED** [review]

**Comment by nyechiel** [comment]
@tpantelis can you please also review before we merge this?

File: src/content/development/security/containers/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: "Container Requirements"
+weight: 50
+---
+
+Current privilege setup is as follows, for non-test containers deployed by Submariner.
+Production containers not described here don’t use extra capabilities.
```
Comment (tpantelis) [inline src/content/development/security/containers/_index.en.md:7]:
```suggestion
The current privilege requirements for non-test containers deployed by Submariner are described below.
Any other production containers not described here don’t require extra privileges.
```

File: src/content/development/security/reporting/_index.en.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+weight: 40
+---
+
+Submariner welcomes and appreciates responsible disclosure of security vulnerabilities.
+
+If you know of a security issue with Submariner, please report it to submariner-security@googlegroups.com.
+
+Submariner aspires to follow the [Kubernetes security reporting process](https://kubernetes.io/docs/reference/issues-security/security/),
+but is far too small of a project to implement those practices. Where applicable, Submariner will follow the principles of the Kubernetes
+process.
```
Comment (tpantelis) [inline src/content/development/security/reporting/_index.en.md:12]:
```suggestion
Submariner aspires to follow the [Kubernetes security reporting process](https://kubernetes.io/docs/reference/issues-security/security/) where possible
but is too small of a project to fully implement those practices.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: src/content/development/security/reporting/_index.en.md (hunk 2 above)
Comment (dfarrell07) [reply src/content/development/security/reporting/_index.en.md:12]: This isn't part of this PR IMHO, it's copied over from the previous file. I also prefer the original version. I think it's a substantial loss to not say "Submariner will follow the principles of the Kubernetes process" - that's basically the core idea.

Comment (tpantelis) [reply src/content/development/security/reporting/_index.en.md:12]: If one equates "_Submariner aspires to follow_..." with "_Submariner will follow_..." then nothing is lost - I've merely included the "_where applicable_" part (as "_where possible_") in the former sentence to render the latter redundant. However if one thinks meaning is lost by losing the "_will follow_" phrasing, then we can change it to 

```suggestion
Submariner follows the [Kubernetes security reporting process](https://kubernetes.io/docs/reference/issues-security/security/) 
where possible but is too small of a project to fully implement those practices.
```

Comment (dfarrell07) [reply src/content/development/security/reporting/_index.en.md:12]: What changes is the nuance that if you follow the link and read the K8s process, it's clearly built for a very large project. The original basically says "Although we'd like to, we're too small to follow the massive process in K8s, but we will follow the principles". The second basically says "We will try to follow the K8s process but can't implement parts of it". I think those are substantially different things.

Comment (tpantelis) [reply src/content/development/security/reporting/_index.en.md:12]: ok

File: src/content/development/security/containers/_index.en.md (hunk 1 above)
Comment (skitt) [reply src/content/development/security/containers/_index.en.md:7]: As I tried to explain in the commit message, this PR describes the current settings, it doesn’t describe the requirements (which are less than the current settings).

**Review by tpantelis — APPROVED** [review]

**Pull request merged**